bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -61,68 +74,121 @@ public void initialize() {
-            updateState(CHANNEL_ENERGY_OUT_L1, energyMeter.getEnergyOutL1());
-
-            updateState(CHANNEL_POWER_IN_L2, energyMeter.getPowerInL2());
-            updateState(CHANNEL_POWER_OUT_L2, energyMeter.getPowerOutL2());
-            updateState(CHANNEL_ENERGY_IN_L2, energyMeter.getEnergyInL2());
-            updateState(CHANNEL_ENERGY_OUT_L2, energyMeter.getEnergyOutL2());
-
-            updateState(CHANNEL_POWER_IN_L3, energyMeter.getPowerInL3());
-            updateState(CHANNEL_POWER_OUT_L3, energyMeter.getPowerOutL3());
-            updateState(CHANNEL_ENERGY_IN_L3, energyMeter.getEnergyInL3());
-            updateState(CHANNEL_ENERGY_OUT_L3, energyMeter.getEnergyOutL3());
-
-            if (getThing().getStatus().equals(ThingStatus.OFFLINE)) {
-                updateStatus(ThingStatus.ONLINE);
+        long currentUpdate = System.currentTimeMillis();
+        if (updateInterval > currentUpdate - lastUpdate) {
+            logger.trace("Update is to early {}, waiting to {}", currentUpdate - lastUpdate, updateInterval);
+            if (!refresh.get()) {
+                return;
             }
-        } catch (IOException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
         }
+        lastUpdate = currentUpdate;
+
+        logger.debug("Update SMAEnergyMeter {} data '{}'", serialNumber, getThing().getUID());
+        updateStatus(ThingStatus.ONLINE);
+
+        updateState(CHANNEL_POWER_IN, energyMeter.getPowerIn());
+        updateState(CHANNEL_POWER_OUT, energyMeter.getPowerOut());
+        updateState(CHANNEL_ENERGY_IN, energyMeter.getEnergyIn());
+        updateState(CHANNEL_ENERGY_OUT, energyMeter.getEnergyOut());
+
+        updateState(CHANNEL_POWER_IN_L1, energyMeter.getPowerInL1());
+        updateState(CHANNEL_POWER_OUT_L1, energyMeter.getPowerOutL1());
+        updateState(CHANNEL_ENERGY_IN_L1, energyMeter.getEnergyInL1());
+        updateState(CHANNEL_ENERGY_OUT_L1, energyMeter.getEnergyOutL1());
+
+        updateState(CHANNEL_POWER_IN_L2, energyMeter.getPowerInL2());
+        updateState(CHANNEL_POWER_OUT_L2, energyMeter.getPowerOutL2());
+        updateState(CHANNEL_ENERGY_IN_L2, energyMeter.getEnergyInL2());
+        updateState(CHANNEL_ENERGY_OUT_L2, energyMeter.getEnergyOutL2());
+
+        updateState(CHANNEL_POWER_IN_L3, energyMeter.getPowerInL3());
+        updateState(CHANNEL_POWER_OUT_L3, energyMeter.getPowerOutL3());
+        updateState(CHANNEL_ENERGY_IN_L3, energyMeter.getEnergyInL3());
+        updateState(CHANNEL_ENERGY_OUT_L3, energyMeter.getEnergyOutL3());
+
+        refresh.set(false);
+    }
+
+    @Override
+    public void run() {
+        long currentTimeMillis = System.currentTimeMillis();
+        if (currentTimeMillis - lastUpdate > updateInterval * 2) {
+            long missedWindow = TimeUnit.MILLISECONDS.toSeconds((updateInterval * 2));
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.GONE,
```

Comment:
ThingStatusDetail.OFFLINE.GONE is probably not appropriate. Rather simply COMMUNICATION_ERROR ?